switchdev_utils/config-service/switchdev_network_config

```diff
@@ -0,0 +1,548 @@
+        raise e
+
+
+def if_up_interface(device):
+    logger.info("Running /sbin/ifup %s" % device)
+    try:
+        run_command('/sbin/ifup', device)
+    except Exception as e:
+        logger.error("Failed to ifup  %s" % (device, str(e)))
+        raise e
+
+
+def bring_interfaces_up(net_interfaces_list):
+    logger.info("Bringing interfaces down")
+    for interface in net_interfaces_list:
+        if_down_interface(interface)
+    logger.info("Bringing interfaces up")
+    for interface in net_interfaces_list:
+        if_up_interface(interface)
+
+
+def validate_config(sriov_pfs, bonds, bridges):
+    sriov_pfs_list = []
+    bonds_list = []
+    for sriov_pf in sriov_pfs:
+        if sriov_pf.get("name"):
+            sriov_pfs_list.append(sriov_pf.get("name"))
+        else:
+            err = "sriov pf name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        try:
+            if int(sriov_pf.get("numOfVfs")):
+                continue
+        except ValueError as value_err:
+            logger.error("numOfVfs for %s must be number" %
+                         sriov_pf.get("name"))
+            raise value_err
+        except TypeError as type_err:
+            logger.error("numOfVfs for %s must be provided" %
+                         sriov_pf.get("name"))
+    for bond in bonds:
+        if bond.get("name"):
+            bonds_list.append(bond.get("name"))
+        else:
+            err = "bond name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        if not bond.get("bondingOptions"):
+            err = "bondingOptions must be provided for bond %s" %\
+                  bond.get("name")
+            logger.error(err)
+            raise ValueError(err)
+        if (bond.get("slaves") and
+                (set(bond.get("slaves")).issubset(set(sriov_pfs_list)))):
+            continue
+        else:
+            err = "slaves of bond %s must be list of sriov_pfs" % \
+                  bond.get("name")
+            logger.error(err)
+            raise ValueError(err)
+    for bridge in bridges:
+        if not bridge.get("name"):
+            err = "bridge name must be provided"
+            logger.error(err)
+            raise ValueError(err)
+        if (bridge.get("ports") and
+                (set(bridge.get("ports")).issubset(set(sriov_pfs_list +
+                                                   bonds_list)))):
+            continue
+        else:
+            err = "ports of bridge %s must be list of sriov_pfs or bonds" % \
+                  bridge.get("name")
+            logger.error(err)
+            raise ValueError(err)
+
+
+def main(argv):
+    config = get_config()
+    sriov_pfs = config.get("pfs")
+    bonds = config.get("linuxBonds")
+    bridges = config.get("ovsBridges")
+    hw_offload_enabled = config.get("hwOffloadEnabled")
+    validate_config(sriov_pfs, bonds, bridges)
```

Comment:
validate_config should do what it says validate the configuration file, IMO it should run after L509 and get `config` as input.

one of its validations should be the config version, and fail if `configVersion.major` > 1
also emit warning if `configVersion.minor` > 0 as it may indicate new optional functionality this script does not know how to handle.